src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -121,8 +121,15 @@ define(function (require, exports, module) {
                 modifierState = modifierStore.getState(),
                 descendants = modifierState.alt;
 
-            this.getFlux().actions.layers.setGroupExpansion(this.props.document, layer,
-                !layer.expanded, descendants);
+            // The clicked layer may an have out-of-date document models due to
+            // the aggressive SCU method in LayersPanel.
+            var documentID = this.props.document.id,
+                documentStore = this.getFlux().store("document"),
+                currentDocument = documentStore.getDocument(documentID),
+                currentLayer = currentDocument.layers.byID(this.props.layer.id);
+
+            this.getFlux().actions.layers.setGroupExpansion(currentDocument, currentLayer,
```

Comment:
This is the same workaround as in `_handleLayerClick` for the problem of stale documents as a result of the parent component's aggressive `shouldComponentUpdate` method. This is not great, but solving it in a nicer way is a task I'd like to delay.